### PR TITLE
feat: support annotation of response-rewrite

### LINKED ISSUE #575: feat: add response-rewrite for Ingress resource
I think this is useful.

xref: https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/response-rewrite.md

### PR DESCRIPTION
**Pull request by An-DJ**
### Type of change:

- [ ] Bugfix
- [X] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

Support 4 annotations of response-rewrite:
* k8s.apisix.apache.org/enable-response-rewrite
* k8s.apisix.apache.org/response-rewrite-status-code
* k8s.apisix.apache.org/response-rewrite-body
* k8s.apisix.apache.org/response-rewrite-body-base64

Close #575

### Pre-submission checklist:

* [X] Did you explain what problem does this PR solve? Or what new features have been added?
* [X] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [X] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by An-DJ** [comment]
@tao12345666333 PTAL

**Comment by tao12345666333** [comment]
Thanks

File: test/e2e/suite-annotations/response_rewrite.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+	"github.com/stretchr/testify/assert"
+
+	"github.com/apache/apisix-ingress-controller/test/e2e/scaffold"
+)
+
+// suite-annotations: response-rewrite annotations
+var _ = ginkgo.Describe("suite-annotations: response-rewrite", func() {
+	s := scaffold.NewDefaultScaffold()
+
+	ginkgo.It("enable in ingress networking/v1", func() {
```
Comment (tao12345666333) [inline test/e2e/suite-annotations/response_rewrite.go:32]: Please add test case to cover `networking/v1beta1` and `extensions/v1beta1`. Although they are old, we still support their use

File: pkg/providers/ingress/translation/annotations/plugins/response_rewrite.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/apache/apisix-ingress-controller/pkg/providers/ingress/translation/annotations"
+	apisixv1 "github.com/apache/apisix-ingress-controller/pkg/types/apisix/v1"
+	"strconv"
+)
+
+type responseRewrite struct{}
+
+// NewResponseRewriteFHandler creates a handler to convert annotations about
+// ResponseRewrite to APISIX response-rewrite plugin.
+func NewResponseRewriteFHandler() PluginAnnotationsHandler {
```
Comment (lingsamuel) [inline pkg/providers/ingress/translation/annotations/plugins/response_rewrite.go:27]: What does the F mean here?

**Review by lingsamuel — APPROVED** [review]

File: test/e2e/suite-annotations/response_rewrite.go (hunk 1 above)
Comment (An-DJ) [reply test/e2e/suite-annotations/response_rewrite.go:32]: > Please add test case to cover `networking/v1beta1` and `extensions/v1beta1`. Although they are old, we still support their use

Done.

File: pkg/providers/ingress/translation/annotations/plugins/response_rewrite.go (hunk 2 above)
Comment (An-DJ) [reply pkg/providers/ingress/translation/annotations/plugins/response_rewrite.go]: > What does the F mean here?

Sorry for my spelling mistake... It has been corrected in the new commit.

**Comment by An-DJ** [comment]
> Please fix the linter errors.

Done. It's like the first-time contributor can't actively trigger the test workflows, could you approve them? : )

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**